src/loaders/FileLoader.js

```diff
@@ -55,225 +55,190 @@ class FileLoader extends Loader {
 
-						} else {
+					// periodically read data into the new stream tracking while download progress
+					return new ReadableStream( {
+						start( controller ) {
 
-							response = view.buffer;
+							readData();
 
-						}
+							function readData() {
 
-						break;
+								reader.read()
+									.then( ( { done, value } ) => {
 
-					case 'document':
+										if ( done ) {
 
-						const parser = new DOMParser();
-						response = parser.parseFromString( data, mimeType );
+											controller.close();
 
-						break;
+										} else {
 
-					case 'json':
+											loaded += value.byteLength;
 
-						response = JSON.parse( data );
+											const event = new ProgressEvent( 'progress', { lengthComputable: true, loaded, total } );
+											for ( let i = 0, il = callbacks.length; i < il; i ++ ) {
 
-						break;
+												const callback = callbacks[ i ];
+												if ( callback.onProgress ) callback.onProgress( event );
 
-					default: // 'text' or other
+											}
 
-						response = data;
+											controller.enqueue( value );
+											readData();
 
-						break;
+										}
 
-				}
+									} );
 
-				// Wait for next browser tick like standard XMLHttpRequest event dispatching does
-				setTimeout( function () {
+							}
 
-					if ( onLoad ) onLoad( response );
+						}
+        	} );
 
-					scope.manager.itemEnd( url );
+				} else {
 
-				}, 0 );
+					const callbacks = loading[ url ];
+					delete loading[ url ];
 
-			} catch ( error ) {
+					for ( let i = 0, il = callbacks.length; i < il; i ++ ) {
 
-				// Wait for next browser tick like standard XMLHttpRequest event dispatching does
-				setTimeout( function () {
+						const callback = callbacks[ i ];
+						if ( callback.onError ) callback.onError( response.statusText );
 
-					if ( onError ) onError( error );
+					}
 
 					scope.manager.itemError( url );
 					scope.manager.itemEnd( url );
 
-				}, 0 );
-
-			}
-
-		} else {
-
-			// Initialise array for duplicate requests
-
-			loading[ url ] = [];
-
-			loading[ url ].push( {
-
-				onLoad: onLoad,
-				onProgress: onProgress,
-				onError: onError
-
-			} );
-
-			request = new XMLHttpRequest();
-
-			request.open( 'GET', url, true );
-
-			request.addEventListener( 'load', function ( event ) {
-
-				const response = this.response;
-
-				const callbacks = loading[ url ];
-
-				delete loading[ url ];
-
-				if ( this.status === 200 || this.status === 0 ) {
-
-					// Some browsers return HTTP Status 0 when using non-http protocol
-					// e.g. 'file://' or 'data://'. Handle as success.
-
-					if ( this.status === 0 ) console.warn( 'THREE.FileLoader: HTTP Status 0 received.' );
-
-					// Add to cache only on HTTP success, so that we do not cache
-					// error response bodies as proper responses to requests.
-					Cache.add( url, response );
+				}
 
-					for ( let i = 0, il = callbacks.length; i < il; i ++ ) {
+			} )
+			.then( async stream => {
```

Comment:
It doesn't look like the "async" keyword is needed here

Reply:
has been removed